veps/sig-compute/10-dra-devices/vep.md

```diff
@@ -0,0 +1,760 @@
+}
+
+type DeviceAttributes struct {
+	// PCIAddress is the PCIe bus address of the allocated device
+	// +optional
+	PCIAddress *string `json:"pciAddress,omitempty"`
+	// MDevUUID is the mediated device uuid of the allocated device
+	// +optional
+	MDevUUID *string `json:"mdevUUID,omitempty"`
+}
+```
+
+The first section vmi.spec.resourceClaims will have a list of devices needed to be allocated for the VM. Having this
+available as a list will allow users to use the device from this list in GPU section or HostDevices section of the
+DomainSpec API.
+
+In v1alpha3 version of [DRA API](https://pkg.go.dev/k8s.io/api@v0.31.0/resource/v1alpha3#DeviceClaim), multiple drivers
+could potentially provision devices that are part of a single claim. For this reason, a separate list of claims required
+for the VMI (section 1) is needed instead of mentioning the resource claim in devices section
+([see Alternate Designs](#Alternative 1))
+
+The second sections allows for the resource claim to be used in the spec.domain.devices section. The two uses cases
+currently handled by the design are:
+
+1. allowing the devices to be used as a gpu device (spec.domain.devices.gpu)
+2. allowing the devices to be used as a host device (spec.domain.device.hostDevices)
+
+The status section of the VMI will contain information of the allocated devices for the VMI when the information is
+available in DRA APIs. The same information will be accessible in virt-handler and virt-launcher. This allows for device
+information to flow from DRA APIs into KubeVirt stack.
+
+Taking a GPU as an example, we can either have a passthrough-GPU as a PCI device or a virtual-GPU as a mediated device.
+The `DeviceAttributes` object will distinguish between these 2 types by populating the appropriate `PCIAddress`/`MDevUUID`
+field for the device identifier. While we haven't mentioned any other device attributes here, this object can be extended
+to hold other device information that may be relevant.
+
+The virt-launcher will have the logic of converting a GPU device into its corresponding domain xml. For device-plugins, it
+will continue to look for env variables (current approach). For DRA devices, it will use the vmi status section to
+generate the domain xml.
+
+### DRA API for reading device related information
+
+The examples below shows the APIs used to generate the vmi.status.deviceStatuses section:
+1. the pod status has reference to the resourceClaimName, pod.status.resourceClaimStatuses[*].resourceClaimName where
+   the name of the claim is same as vmi.spec.resourceClaims[*].Name
+1. pod spec has node name, pod.spec.nodeName
+1. the resourceclaim status has device name and driver use for allocating the device,
+   resourceclaim.status.allocation.devices[*].deviceName and resourceclaim.status.allocation.devices[*].driver, where
+   resourceclaim.status.allocation.devices[*].request is same as vmi.spec.domain.devices[*].gpus[*].claim.request
```

Comment:
The stars above are interpreted as *italic* for some reason.

Reply:
removed the stars, kept `[]` to help readers understand its a list